helpers/focus.js

```diff
@@ -26,14 +26,16 @@ export function getComposedActiveElement() {
 	return node;
 }
 
-export function getFirstFocusableDescendant(node, includeHidden) {
+export function getFirstFocusableDescendant(node, includeHidden, predicate) {
+	if (predicate === undefined) predicate = () => true;
+
 	const composedChildren = getComposedChildren(node);
 
 	for (let i = 0; i < composedChildren.length; i++) {
-		if (isFocusable(composedChildren[i], includeHidden)) return composedChildren[i];
+		if (isFocusable(composedChildren[i], includeHidden) && predicate(composedChildren[i])) return composedChildren[i];
 
-		const focusable = getFirstFocusableDescendant(composedChildren[i], includeHidden);
-		if (focusable) return focusable;
+		const focusable = getFirstFocusableDescendant(composedChildren[i], includeHidden, predicate);
+		if (focusable && predicate(focusable)) return focusable;
```

Comment:
Is the `predicate(focusable)` needed here? -it looks like this will already have been invoked inside the `getFirstFocusableDescendant` call on the previous line.  I _think_ we just need to check if an element was returned so that we don't return `null` below.

Reply:
You're right, thanks! The predicate will have returned true for the found descendant in line 35 and then bubbled up.